From 1ec06f004f03e39aa74432ee2d2b577fe131d106 Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Fri, 1 Mar 2019 21:02:56 +0100
Subject: [PATCH 2/6] Overview: fix gtk.show_uri() call

show_uri() crashes if passed gdk.Screen(), but works with
gdk.Screen().get_default(). This isn't seen in current hamster
because the os.path.split() call causes a NameError exception
first. But it's unclear anyway why the path was split (and only the
first directory component used).

Signed-off-by: Martin Wilck <mwilck@suse.com>
---
 src/hamster/overview.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/hamster/overview.py b/src/hamster/overview.py
index 909d065..fa0d808 100755
--- a/src/hamster/overview.py
+++ b/src/hamster/overview.py
@@ -557,7 +557,8 @@ class Overview(Controller):
                 webbrowser.open_new("file://%s" % path)
             else:
                 try:
-                    gtk.show_uri(gdk.Screen(), "file://%s" % os.path.split(path)[0], 0)
+                    gtk.show_uri(gdk.Screen().get_default(),
+                                 "file://%s" % path, gdk.CURRENT_TIME)
                 except:
                     pass # bug 626656 - no use in capturing this one i think
 
-- 
2.20.1

